src/EnsembleKalmanInversions.jl

```diff
@@ -72,6 +73,17 @@ function inverse_covariance_transform(cn::Tuple{Vararg{ConstrainedNormal}}, para
     dT = Diagonal(@. -(upper_bound - lower_bound) * exp(parameters) / (1 + exp(parameters))^2)
     return dT * covariance * dT'
 end
+=#
+
+function inverse_covariance_transform(Π, parameters, covariance)
+    diag = [covariance_transform_diagonal(Π[i], parameters[i]) for i=1:length(Π)]
+    dT = Diagonal(diag)
+    return dT * covariance * dT'
+end
+
+covariance_transform_diagonal(::LogNormal, p) = exp(p)
+covariance_transform_diagonal(::Normal, p) = I
+covariance_transform_diagonal(Π::ConstrainedNormal, p) = - (Π.upper_bound - Π.lower_bound) * exp(p) / (1 + exp(p))^2
```

Comment:
I changed `p` to `I` on line 85 and `(1 + exp(p)^2)` to `(1 + exp(p))^2` on line 86

Reply:
And yeah now I see that returning `p` was wrong.

Reply:
Yes good point, let's change to `1`